### ISSUE TITLE
Jinja templater should read variables from `dbt_project.yml` when `apply_dbt_builtins = true`

### ISSUE THREAD

**Issue by edmondop** [sqlfluff/sqlfluff#7304]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

## Problem
The Jinja templater supports `apply_dbt_builtins = true` to provide dbt macro stubs, but doesn't read variables from `dbt_project.yml`. This forces manual duplication of all dbt variables in `.sqlfluff` config.

## Use Case
Many environments can't connect to the warehouse (CI runners, security restrictions, no VPN access), making the dbt templater unusable. The Jinja templater is the only option, but variables must be manually synced.

## Current Config
```ini
[sqlfluff]
templater = jinja

[sqlfluff:templater:jinja]
apply_dbt_builtins = true
load_macros_from_path = macros
```

This fails on `{{ var('my_variable') }}` because variables aren't loaded.

## Inconsistency
If `apply_dbt_builtins` acknowledges dbt usage, why not read variables from `dbt_project.yml`? **Reading a YAML file doesn't require database connectivity** - it's literally this simple:

```python
import yaml
from pathlib import Path

# Load dbt variables
dbt_project_path = Path("dbt_project.yml")
with open(dbt_project_path) as f:
    dbt_config = yaml.safe_load(f)

vars = dbt_config.get('vars', {})
# Add to Jinja context
```

## Proposed Solution
Add option to auto-load variables:

```ini
[sqlfluff:templater:jinja]
apply_dbt_builtins = true
load_dbt_project_vars = true  # New option
```

Would search for `dbt_project.yml` and load the `vars` section into Jinja context.

### Dialect

None, generic feature

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by mauro-lanza** [comment]
Sounds interesting, i also have mostly the same to load the variables, but then also need a separate

```python
# macro_lib_mocks/__init__.py
def get_var(var_name):
    """Return a dbt project variable by name."""
    return all_vars[var_name]
```

and

```jinja
# macros/get_var.sql
{% macro get_var(var_name) -%}
  {{ return(var(var_name)) }}
{%- endmacro %}
```

So in a model i can do

```jinja
# random model
{% set columns = get_var('var_list_columns') %}
select
    id
    {% for column in columns %}
      ,{{ column }}
    {% endfor %}
from {{ ref('table') }}
```

That way it works if you happen to use both jinja and dbt templater and can get by with an sqlfluff config like:

```ini
# ---------------------------------------------------------------------------
# TEMPLATER CONFIGURATION (DBT)
# ---------------------------------------------------------------------------
[sqlfluff:templater:dbt]
apply_dbt_builtins = True
load_macros_from_path = macros

[sqlfluff:templater:dbt:context]
# Custom variable for is_incremental() to evaluate as True during linting
sqlfluff_treat_is_incremental_as_true = True

# ---------------------------------------------------------------------------
# TEMPLATER CONFIGURATION (JINJA)
# ---------------------------------------------------------------------------
[sqlfluff:templater:jinja]
apply_dbt_builtins = True
load_macros_from_path = macros,macro_lib_mocks/macro_overrides
exclude_macros_from_path = macros/get_var.sql
library_path = macro_lib_mocks

[sqlfluff:templater:jinja:context]
run_started_at = datetime.datetime(2025, 1, 1, 4, 0, 0)
```
